jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java

```diff
@@ -1198,6 +1198,13 @@ public void testUpgradeJDK() throws CommandFailureException, IOException {
 		when(Config.mockResourceService.getResourceContent(resourceIdentifier)).thenReturn(resourceContent);
 		Jvm response = jvmService.upgradeJDK(mockJvm.getJvmName(), Config.mockUser);
 		assertEquals(response.getJvmName(), mockJvm.getJvmName());
+
+		/**
+		 * The persistence Service will be called three times - to fetch JVM details by
+		 * NAME-To fetch Setenv.bat Resource content - To generate the resource file-
+		 * Setenv.bat . JDK upgrade is assumed to be successful,if all these three steps
+		 * happens without error
+		 **/
 		verify(Config.mockJvmPersistenceService, times(3)).findJvmByExactName(anyString());
```

Comment:
There are other verify statements missing.

When you go through the upgradeJDK method there are interactions with other services (that are correctly mocked for this test case) that need to be verified. This is making sure that we have automated code coverage for the core feature at the JvmService level.

- the JvmControlService can be verified to make sure it checks for a STOPPED state, and that the Windows/Linux service is deleted and then created again later
 - the BinaryDistributionService can be verified to make sure that it is distributing the JDK media

The test doesn't need to physically perform these steps (i.e. actually pushing files to the file system), but it does need to verify that the mocked helper services that would perform those functions are being called as expected.

Reply:
These changes were addressed